Microsoft.Recognizers.Text.DateTime/Extractors/BaseMergedExtractor.cs

```diff
@@ -82,6 +98,19 @@ private void AddTo(List<ExtractResult> dst, List<ExtractResult> src)
             return config.FromToRegex.IsMatch(er.Text);
         }
 
+        private bool FilterAmbiguousSingleWord(ExtractResult er, string text)
+        {
+            if (config.SingleAmbiguousMonthRegex.IsMatch(er.Text.ToLowerInvariant()))
+            {
+                var stringBefore = text.Substring(0, (int) er.Start).TrimEnd();
+                if (!config.PrepositionSuffixRegex.IsMatch(stringBefore))
```

Comment:
There should be more negative unit tests to be sure this filtering doesn't break common scenarios.